watertap/examples/flowsheets/mvc/mvc_single_stage.py

```diff
@@ -0,0 +1,745 @@
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+from pyomo.environ import (
+    ConcreteModel,
+    value,
+    Constraint,
+    Objective,
+    Var,
+    TransformationFactory,
+    units as pyunits,
+    check_optimal_termination,
+)
+from pyomo.network import Arc
+
+import pyomo.environ as pyo
+import pyomo.util.infeasible as infeas
+from idaes.core import FlowsheetBlock
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import propagate_state
+from idaes.core.util.model_diagnostics import DegeneracyHunter
+from idaes.models.unit_models import Feed, Separator, Mixer, Product
+from idaes.models.unit_models.translator import Translator
+from idaes.models.unit_models.separator import SplittingType
+from idaes.models.unit_models.mixer import MomentumMixingType
+from idaes.models.unit_models.heat_exchanger import (
+    HeatExchanger,
+    HeatExchangerFlowPattern,
+)
+from idaes.core import UnitModelCostingBlock
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+
+from watertap.unit_models.mvc.components import Evaporator, Compressor, Condenser
+from watertap.unit_models.mvc.components.lmtd_chen_callback import (
+    delta_temperature_chen_callback,
+)
+from watertap.unit_models.pressure_changer import Pump
+import watertap.property_models.seawater_prop_pack as props_sw
+import watertap.property_models.water_prop_pack as props_w
+from watertap.costing import WaterTAPCosting
+import math
+
+
+def main():
+    # build, set operating conditions, initialize for simulation
+    m = build()
+    set_operating_conditions(m)
+    add_Q_ext(m, time_point=m.fs.config.time)
+    initialize_system(m)
+    scale_costs(
+        m
+    )  # rescale costs after initialization because scaling depends on flow rates
+    fix_outlet_pressures(m)  # outlet pressure are initially unfixed for initialization
+
+    # set up for minimizing Q_ext in first solve
+    print(
+        "DOF after initialization: ", degrees_of_freedom(m)
+    )  # should be 1 because Q_ext is unfixed
+    m.fs.objective = Objective(expr=m.fs.Q_ext[0])
+
+    print("\n***---First solve - simulation results---***")
+    solver = get_solver()
+    results = solve(m, solver=solver, tee=False)
+    print("Termination condition: ", results.solver.termination_condition)
+    display_metrics(m)
+    display_design(m)
+
+    print("\n***---Second solve - optimization results---***")
+    m.fs.Q_ext[0].fix(0)  # no longer want external heating in evaporator
+    del m.fs.objective
+    set_up_optimization(m)
+    results = solve(m, solver=solver, tee=False)
+    print("Termination condition: ", results.solver.termination_condition)
```

Comment:
You should probably change to assert optimal termination to more clearly alert the user to a failed solve.